e2e/runner/inscription.go

```diff
@@ -0,0 +1,116 @@
+package runner
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"io"
+	"net/http"
+)
+
+type commitResponse struct {
+	Address string `json:"address"`
+}
+
+type revealResponse struct {
+	RawHex string `json:"rawHex"`
+}
+
+type revealRequest struct {
+	Txn     string `json:"txn"`
+	Idx     int    `json:"idx"`
+	Amount  int    `json:"amount"`
+	FeeRate int    `json:"feeRate"`
+	To      string `json:"to"`
+}
+
+// InscriptionBuilder is a util struct that help create inscription commit and reveal transactions
+type InscriptionBuilder struct {
+	sidecarUrl string
+	client     http.Client
+}
+
+func (r *InscriptionBuilder) GenerateCommitAddress(memo []byte) (string, error) {
+	// Create the payload
+	postData := map[string]string{
+		"memo": hex.EncodeToString(memo),
+	}
+
+	// Convert the payload to JSON
+	jsonData, err := json.Marshal(postData)
+	if err != nil {
+		return "", err
+	}
+
+	postUrl := r.sidecarUrl + "/commit"
+	req, err := http.NewRequest("POST", postUrl, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return "", errors.Wrap(err, "cannot create commit request")
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	// Send the request
+	resp, err := r.client.Do(req)
+	if err != nil {
+		return "", errors.Wrap(err, "cannot send to sidecar")
+	}
+	defer resp.Body.Close()
+
+	// Read the response body
+	var response commitResponse
+	err = json.NewDecoder(resp.Body).Decode(&response)
+	if err != nil {
+		return "", err
+	}
+
+	fmt.Print("raw commit response ", response.Address)
+
+	return response.Address, nil
+}
+
+func (r *InscriptionBuilder) GenerateRevealTxn(to string, txnHash string, idx int, amount float64) (string, error) {
```

Comment:
Is it possible to add comment to explain method

Reply:
added